### ISSUE TITLE
[TextInputLayout] Screenreaders can announce text hidden behind bullets while in Password mode

### ISSUE THREAD

**Issue by cpanga** [material-components/material-components-android#2300]
**Description:** 

When the TextInputLayout is using `endIconMode="password_toggle"` such that it is operating as a password entry field - a screenreader can read the text that is supposed to be hidden behind the bullets.

This only happens when the screenreader is navigating by 'Words' - navigating normally by swiping left or right will only announce the amount of characters while the password is hidden

Repro scenario:

- Go to talkback settings
  - Navigate to `Customize menus`->`Customize reading controls`->`Words`
  - Ensure 'Words' is checked
- Go to a screen with a TextInputlayout in password mode:
  - Ensure the password is hidden
  - Navigate by words (Swipe down then up in a V pattern to change which reading control you're using, then swipe up/down to navigate by this reading mode)
  - Navigate through the TextInputLayout - the password is read out by the screenreader while it should be hidden

**Expected behavior:** The screenreader should not read out the text while the password is hidden

Password not hidden:

![2021-07-14_14h34_29](https://user-images.githubusercontent.com/87429484/125631096-f929c017-1601-469a-bfe1-3e567d36d870.png)

Password hidden:

![image](https://user-images.githubusercontent.com/87429484/125631395-22238563-b983-4c8f-a7d2-da3377d46167.png)

**Source code:** Standard implementation of TextInputLayout with `endIconMode="password_toggle"`

**Android API version:** 30

**Material Library version:** 1.4.0

**Device:** Pixel XL on Android 10

I've also attempted repro on a couple other apps just to confirm it's not specific to our implementation
 - LinkedIn
 - Spotify
 - Slack
All suffer from the same issue

**Comment by leticiarossi** [comment]
This sounds like a TalkBack issue if it happens in other apps, not a MDC library issue. However, I'm not able to reproduce this in our catalog. The announcements follow your expected behavior when navigating by words.

If you're still seeing this issue using the latest version of the library and of TalkBack then please reopen this bug

**Comment by pramitpatel** [comment]
I am facing similar issue currently. 

MDS version 1.6.0

**Description-**
When the TextInputLayout is using endIconMode="password_toggle" such that it is operating as a password entry field - If you connect any ui testing tool like Appium it can read the password that is supposed to be hidden behind the DOTS.

I did some investigation and reach to conclusions that whenever I am using combination of TextInputLayout and TextInputEditText this issue occurs. 

When I use only TextInputEditText and set PasswordTransformationMethod to transformationMethod property. Password value is hidden even when checking with Appium tool. 

**Repro scenario:**

- Go to a screen with a TextInputlayout in password mode:
- Type password and ensure the password is hidden.
- Connect device/emulator with Appium Inspector tool.
- Select Password field and check attributes you would be able to see plain text password even though it is hidden on the screen.

**Expected behavior:** 
- Appium or any other tool should not be able to read the text while the password is hidden. 
- TextInputLayout should not change TextInputEditText behaviour when used in combination for password field.
